File: Olympiad/COCI/official/2008/contest1_solutions/prinova.pas
program prinova;

var
   i, j, n : longint;
   p : array[1..100] of longint;
   A, B : longint;
   rjesenje : longint;

function racunaj( x : longint ) : longint;
var
   i, ret : longint;
begin
   if x mod 2 = 0 then begin
      racunaj := 0;
   end else begin

      ret := 1000000000;
      for i := 1 to n do
         if abs( x-p[i] ) < ret then
            ret := abs( x-p[i] );

      racunaj := ret;
   end;
end;

procedure probaj( x : longint );
begin
   if (x >= A) and (x <= B) then
      if racunaj(x) > racunaj(rjesenje) then
         rjesenje := x;
end;

begin
   readln( n );

   for i := 1 to n do
      read( p[i] );
   readln;
   readln( A, B );

   rjesenje := 0;

   probaj( A );
   probaj( A + 1 );
   probaj( B );
   probaj( B - 1 );

   for i := 1 to n-1 do
      for j := i+1 to n do begin
         probaj( (p[i]+p[j]) div 2 - 1 );
         probaj( (p[i]+p[j]) div 2 );
         probaj( (p[i]+p[j]) div 2 + 1 );
      end;

   writeln( rjesenje );
end.

File: Olympiad/COCI/official/2008/contest1_solutions/prinova.cpp
#include <cstdio>

int n;
int p[100];
int A, B;
int rjesenje;

int abs( int x ) { return x < 0 ? -x : x; }

int racunaj( int x ) {
   if( x%2 == 0 ) return 0;

   int ret = 1000000000;
   for( int i = 0; i < n; ++i ) 
      ret <?= abs( x-p[i] );
   return ret;
}

void probaj( int x ) {
   if( x < A || x > B ) return;

   if( racunaj( x ) > racunaj( rjesenje ) ) rjesenje = x;   
}


int main( void ) {
   scanf( "%d", &n );
   for( int i = 0; i < n; ++i ) scanf( "%d", &p[i] );
   scanf( "%d%d", &A, &B );

   rjesenje = 0;

   probaj( A ); 
   probaj( A+1 );
   probaj( B ); 
   probaj( B-1 );

   for( int i = 0; i < n; ++i ) 
      for( int j = i+1; j < n; ++j ) {
         probaj( (p[i]+p[j])/2-1 );
         probaj( (p[i]+p[j])/2 );
         probaj( (p[i]+p[j])/2+1 );
      }

   printf( "%d\n", rjesenje );

   return 0;
}
